README.md

```diff
@@ -3,10 +3,71 @@ API and storage for logging event and audit messages, persisted in JSON
 
+The logserver serves as the **Audit Record Repository**, as detailed in the 
+Basic Audit Log Patterns
+[(BALP)](https://profiles.ihe.net/ITI/BALP/volume-1.html#1-52-basic-audit-log-patterns) 
+implementation guide.
+
 ## Event Schema
-**logserver** is agnostic to the format, provided it's valid JSON.  Any number
-of database tables can be used, but only the single **"events"** table is built
-in, containing a PostgreSQL JSONB column,  **"event"**.
+**logserver** is agnostic to the format, provided it is valid JSON.  Any number
+of database tables can be used, but only the single `events` table is built
+in, containing a PostgreSQL JSONB column,  `event`.
+
+It is desirable to generate log events complaint with the [FHIR audit event](
+https://www.hl7.org/fhir/auditevent.html) resource.  To generate an AuditEvent
+resource nested within each `event` is cumbersome, however the following field
+parity is recommended:
+
+NB: all fields are considered optional unless marked as **required**
+
+- `severity`: **required** element.  Use built in log level rather than adding
+  an additional `severity` field:
+  - `critical`: Critical condition with application.  Include `emergency` and
+    `alert` levels in this category.
+  - `error`: Error condition with the application.
+  - `warning`: Warning needing attention before escalation to error.
+  - `info`: Normal operational messages not requiring action.  Include
+    `notice` level in this category (normal but significant).
+  - `debug`: Debug level messages, useful to application developers.
+- `action`: **required** element to describe the type of operation performed.
+  - `C`: create - creating a new resource, such as adding a patient.
+  - `R`: read/view/search - data retrieved or viewed w/o modification.
+  - `U`: update - indicates existing data was modified.
+  - `D`: delete - indicates data was removed or deleted.
+  - `E`: execute - system or application function such as log-on, program
+    execution or perform a query/search.
+- `occurred`: Use built in log message time rather than adding an additional
+  `occurred` field.
+- `category`: major type of the event, such as:
+  - `authentication`: events related to login or authentication.
+  - `authorization`: events related to access control or permissions changes.
+  - `security`: general security-related events.
+  - `data-access`: events where healthcare data is accessed or modified.
+  - `configuration`: events involving system or configuration changes.
+- `code`: specific type of event.  See [audit-event-sub-type valueset](
+  https://www.hl7.org/fhir/valueset-audit-event-sub-type.html) for full list.
+  - `login`
+  - `create`
+  - `read`
+  - `update`
+  - `delete`
+  - `search`
+- `patient`: the **subject** of the activity, i.e. `Patient/ab-123-ef`
+- `agent`: actor involved in the event, i.e. `Practitioner/123-abc`
+- `source`: event reporter or system generating the audit event including version
```

Comment:
I'd be interested in seeing `version` as a top-level attribute. No strong feelings, I just worry about `version` being just as optional as `source` (and possibly omitted more often)